File: leetcode_C++/p2437.cpp
/*
@Author: mrxhar
@Date:   2023-09-07 09:38:48
@Email:  1337904145@qq.com
@Blog:lid-clock-times/?envType=daily-question&envId=2023-09-07

*/
#include <string>
using std::string;

class Solution {
public:
    int countTime(string time) {
        // 小时和分钟是分开来计算的，小时不影响分钟
        int hour = 1, minute = 1;
        // 问号
        // 先计算 hour
        if (time[0] == '?' && time[1] == '?')
            hour = 24;
        else if (time[0] == '?')
            hour = (time[1] - '0') <= 3 ? 3 : 2;
        else
            hour = (time[0] - '0') < 2 ? 10 : 4;
        
        // 计算minute
        minute = (time[3] == '?' ? 6 : 1) * (time[4] == '?' ? 10 : 1);
        return hour * minute;
    }
};

int main() {
    return 0;
}

File: leetcode_py/p2437.py
class Solution:
    def countTime(self, time: str) -> int:
        hour, minute = 1, 1

        if time[0] == '?' and time[1] == '?':
            hour = 24
        elif time[0] == '?':
            if int(time[1]) <= 3:
                hour = 3
            else:
                hour = 2
        elif time[1] == '?':
            if int(time[0]) <= 1:
                hour = 10
            else:
                hour = 4

        if time[3] == '?':
            minute *= 6
        if time[4] == '?':
            minute *= 10
        return hour + minute
